config/tests/samples/create/harness.go

```diff
@@ -484,45 +484,33 @@ func (t *Harness) waitForCRDReady(obj client.Object) {
 	}
 }
 
-func (h *Harness) CompareGoldenFile(p string, got string, normalizers ...func(s string) string) {
-	if os.Getenv("WRITE_GOLDEN_OUTPUT") != "" {
-		// Short-circuit when the output is correct
-		b, err := os.ReadFile(p)
-		if err == nil {
-			want := string(b)
-			for _, normalizer := range normalizers {
-				got = normalizer(got)
-				want = normalizer(want)
-			}
-			if want == got {
-				return
-			}
-		}
-
-		if err := os.WriteFile(p, []byte(got), 0644); err != nil {
-			h.Fatalf("failed to write golden output %s: %v", p, err)
-		}
-		h.Errorf("wrote output to %s", p)
-	} else {
-		want := string(h.MustReadFile(p))
-
-		for _, normalizer := range normalizers {
-			got = normalizer(got)
-			want = normalizer(want)
-		}
-
+func (h *Harness) CompareGoldenFile(p string, got string) {
+	b, err := os.ReadFile(p)
+	if err == nil {
+		h.Logf("golden file %s exist, comparing with it", p)
+		want := string(b)
 		if diff := cmp.Diff(want, got); diff != "" {
 			h.Errorf("unexpected diff in %s: %s", p, diff)
```

Comment:
We do want to write the golden file here also

Reply:
Just my personal preference: people may not 100% understand the command they use and simply copy&paste. My idea was once we have a golden file exists, we do not want users to override it. While, if we feel we need to update the golden file, we can always remove it and regenerate. Maybe I can make the log message more clear: `golden file exists, comparing with it. If you want to update the golden file, please remove the old one and regenerate it by specifying "WRITE_GOLDEN_OUTPUT"`